job_import.py

```diff
@@ -21,12 +23,56 @@
     with open("/sheepdog-creds.json") as pelican_creds_file:
         sheepdog_creds = json.load(pelican_creds_file)
 
-    DB_URL = "jdbc:postgresql://{}/{}".format(
-        sheepdog_creds["db_host"], sheepdog_creds["db_database"]
-    )
+    if "guid" in input_data_json:
+        print("a guid was supplied to the job")
+        print("we are getting a signed url for the given guid")
+
+        host = "https://" + hostname
+
+        auth_headers = {"Authorization": "Bearer "+ access_token}
+
+        api_url = host + "/user/data/download/" + input_data_json["guid"] + "?protocol=s3"
+
+        signed_request = requests.get(api_url, headers=auth_headers)
+        
+        signed_url = signed_request.json()
+        print("the signed url is ", signed_url["url"])
+        input_data_json["url"] = signed_url["url"]
+
+    # DB_URL = "jdbc:postgresql://{}/{}".format(
+    #     sheepdog_creds["db_host"], sheepdog_creds["db_database"]
+    # )
     DB_USER = sheepdog_creds["db_username"]
     DB_PASS = sheepdog_creds["db_password"]
 
+    NEW_DB_NAME = input_data_json["db"]
+
+    # create a database in the name that was passed through
+    engine = sqlalchemy.create_engine("postgres://{user}:{password}@{host}/postgres".format(user=DB_USER, password=DB_PASS, host=sheepdog_creds["db_host"]))
+    conn = engine.connect()
+    conn.execute("commit")
+
+    print("we are creating a new database named ", NEW_DB_NAME)
+
+    create_db_command = "create database " + NEW_DB_NAME
+    print("This is the db create command: ", create_db_command)
+
+    grant_db_access = "grant all on database " + NEW_DB_NAME + " to sheepdog with grant option"
+    print("This is the db access command: ", grant_db_access)
```

Comment:
Using strings directly is not safe for SQL and it's better to rewrite this with Textual SQL: https://docs.sqlalchemy.org/en/13/core/tutorial.html#using-textual-sql

It would look like (not tested):
```
create_db_command = text("create database :db")
```

And run:
```
conn.execute(create_db_command, db= NEW_DB_NAME)
```

Reply:
Added these and tested to make sure sql commands work correctly